### ISSUE TITLE
Re-training Suzuki dataset fails

### ISSUE THREAD

**Issue by cjpcool** [rxn4chemistry/rxn_yields#18]
Hi, thanks for your great work. I have some issues when trying to retrain the model on the Suzuki dataset with different train/val split.

I can successfully run the checkpoint you provided, and the RMSE is great. However, I tried to re-train Bert on Suzuki, the results are super weak. The loss cannot decrease, and the final predicted yields are the same. I used the default hyperparameters provided for training Buchwald-Hartwig, so this also might be the problem of hyperparameters. The prediction figure in evaluate dataset is as follow.
![image](https://github.com/user-attachments/assets/5735bcff-4e50-47c0-bc1b-07805893de9d)

Here is my training script:
~~~
data_path = './data/Suzuki-Miyaura/random_splits/random_split_0.tsv'
output_dir = './trained_model/suzuki_miyaura/random_split_0'

split = 4096
# Data load
df = pd.read_csv(data_path, sep='\t')
# df = df.sample(frac=1)
# df.to_csv()
train_idx = df.index[:split].values
test_idx = df.index[split:].values


train_df = df.iloc[:split][['rxn', 'y']] 
test_df = df.iloc[split:][['rxn', 'y']]
train_df.columns = ['text', 'labels']
test_df.columns = ['text', 'labels']

mean = train_df.labels.mean()
std = train_df.labels.std()
train_df.loc[:,'labels'] = (train_df['labels'] - mean) / std
test_df.loc[:, 'labels'] = (test_df['labels'] - mean) / std

print(train_df['labels'])
print(train_df['labels'].mean(), train_df['labels'].std())

# Load model
model_args = {
     'num_train_epochs': 15, 'overwrite_output_dir': True,
    'learning_rate': 0.00009659, 'gradient_accumulation_steps': 1,
    'regression': True, "num_labels":1, "fp16": False,
    "evaluate_during_training": False, 'manual_seed': 42,
    "max_seq_length": 300, "train_batch_size": 16,"warmup_ratio": 0.00,
    "config" : { 'hidden_dropout_prob': 0.7987 },
    "wandb_project": "rxn_yields",
    "wandb_kwargs": {
        "entity": "cjpcool",
    },
}

model_path =  pkg_resources.resource_filename(
                "rxnfp",
                f"models/transformers/bert_pretrained" # change pretrained to ft to start from the other base model
)

yield_bert = SmilesClassificationModel("bert", model_path, num_labels=1, 
                                       args=model_args, use_cuda=torch.cuda.is_available())

yield_bert.train_model(train_df, output_dir=output_dir, eval_df=test_df)
~~~
Is there any problem in my script? If it might be caused by the unsuitable hyperparameters, could you provide them for re-implementing your results?

I greatly appreciate your brilliant work, and look forward to your response!

**Comment by cjpcool** [comment]
I've resolved this issue by setting "evaluation_during_training" to "true" by referring https://github.com/rxn4chemistry/rxn_yields/issues/7
